endpoints/events/event_test.go

```diff
@@ -750,3 +760,57 @@ func TestReadIntegrationType(t *testing.T) {
 		}
 	}
 }
+
+func TestShouldReturnBadRequestWhenVTypeIsInvalid(t *testing.T) {
+
+	// prepare
+	reqData := ""
+
+	tests := map[string]struct {
+		req                *http.Request
+		expectedStatusCode int
+		expectedStatus     string
+	}{
+		"vtype parameter is missing": {
+			httptest.NewRequest("GET", "/event?t=vast&b=bidID", strings.NewReader(reqData)),
+			400,
+			"invalid request: parameter 'vtype' is required\n",
+		},
+		"invalid vtype parameter": {
+			httptest.NewRequest("GET", "/event?t=vast&vtype=abc&b=bidID", strings.NewReader(reqData)),
+			400,
+			"invalid request: unknown vtype: 'abc'\n",
+		},
+	}
+
+	for _, test := range tests {
+		// mock AccountsFetcher
+		mockAccountsFetcher := &mockAccountsFetcher{}
+
+		// mock PBS Analytics Module
+		mockAnalyticsModule := &eventsMockAnalyticsModule{
+			Fail: false,
+		}
+
+		// mock config
+		cfg := &config.Configuration{
+			AccountDefaults: config.Account{},
+		}
+
+		recorder := httptest.NewRecorder()
+
+		e := NewEventEndpoint(cfg, mockAccountsFetcher, mockAnalyticsModule)
+		// execute
+		e(recorder, test.req, nil)
+
+		d, err := ioutil.ReadAll(recorder.Result().Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// validate
+		assert.Equal(t, test.expectedStatusCode, recorder.Result().StatusCode, "Expected 400 on request with invalid vtype parameter")
```

Comment:
You can replace the string at the end here with the newly added description for each test case.